gems/quilt_rails/README.md

```diff
@@ -119,21 +142,204 @@ function App() {
+- Components
+- Co-located CSS modules
+- Co-located unit tests
+- Test setup files
+
+```
+└── app
+    └── ui
+        ├─- index.js (exports a React component)
+        ├── styles (optional)
+        │   └── settings.scss (global vars and @polaris overrides)
+        │
+        └── tests (optional)
+        │   └── each-test.ts
+        │   └── setup.ts
+        └── features (optional)
+            ├── App
+            │   ├── index.js
+            │   ├── App.js
+            │   └── tests
+            │       └── App.test.js
+            │
+            ├-─ MyComponent
+            │   ├-─ index.js
+            │   ├-─ MyComponent.js
+            │   ├── MyComponent.scss (optional; component-scoped CSS styles, mixins, etc)
+            │   └── tests
+            │       └── MyComponent.test.js
+            │
+            └── sections (optional; container views that compose presentation components into UI blocks)
+                └── Home
+                    ├-─ index.js
+                    └── Home.js
+```
+
+## API
+
+### ReactRenderable
+
+The `ReactRenderable` mixin is intended to be used in Rails controllers, and provides only the `render_react` method. This method handles proxying to a running `@shopify/react-server`.
+
+
+```ruby
+class ReactController < ApplicationController
+  include Quilt::ReactRenderable
+
+  def index
+    render_react
+  end
+end
+```
+
+### Engine
+
+`Quilt::Engine` provides a preconfigured controller which consumes `ReactRenderable` and provides an index route which uses it.
+
+```ruby
+# config/routes.rb
+Rails.application.routes.draw do
+  # ...
+  mount Quilt::Engine, at: '/path/to/react'
+end
+```
+
+### Configuration
+
+The `configure` method allows customization of the address the service will proxy to for UI rendering.
+
+```ruby
+  Quilt.configure do |config|
+    config.react_server_host = "localhost:3000"
+    config.react_server_protocol = 'https'
+  end
+```
+
+### Generators
 
-### `quilt:install`
+#### `quilt:install`
 
-Installs the Node dependencies, provide a basic React app (in TypeScript) and mounts the Quilt engine inside of your `config/routes.rb` file.
+Installs the Node dependencies, provide a basic React app (in TypeScript) and mounts the Quilt engine in `config/routes.rb`.
 
-### `sewing_kit:install`
+#### `sewing_kit:install`
 
 Adds a basic `sewing-kit.config.ts` file.
 
 ## Advanced use
 
-### Interacting with the request / response in your React code
+### Testing
+
+For fast tests with consistent results, test front-end components using the tools provided by sewing-kit instead of Rails integration tests.
+
+Use [`sewing-kit test`](https://github.com/Shopify/sewing-kit/blob/master/docs/commands/test.md#L3) to run all `.test.tsx` files in the `app/ui` directory. [Jest](https://jestjs.io/) is used as a test runner, with customization available via [its sewing-kit plugin](https://github.com/Shopify/sewing-kit/blob/master/docs/plugins/jest.md).
```

Comment:
The flip/flopping between `.ts`/`.tsx` and `js`/`.jsx` might get confusing. Maybe we just have a consistent way of saying all of them.